satip/utils.py

```diff
@@ -845,8 +851,17 @@ def save_to_zarr_to_s3(dataset: xr.Dataset, filename: str):
         log.debug(f"New times for {path}: {new_times}", memory=get_memory())
 
         log.debug(f"Saved to temporary file {path}, now pushing to {filename}", memory=get_memory())
-        # save to s3
-        filesystem = fsspec.open(filename).fs
+
+        # Save to the specified backend
+        filesystem = fsspec.open(
+            filename,
+            target_options={
+                "storage_options": {
+                    "token": STORAGE_BACKEND_TOKEN
```

Comment:
This might be easier, and then the variable could be set later, rather than when this is imported

```suggestion
                    "token": os.getenv('STORAGE_BACKEND_TOKEN')
```